### DISCUSSION TITLE
Logs getting deleted after 7 days

### DISCUSSION THREAD

**Discussion by loganterfehr** [Security-Onion-Solutions/securityonion#12417] [Unsupported Versions]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/101576b6-1978-43f9-8dbe-0aa7007518fa)

Any idea why my logs are getting deleted after around 7 days? I thought that I had the deleted logs set to 120 days?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/56eeb5b8-00e3-46b7-b656-fd4074659afe)

Let me know if there is a different area that could be doing this and not just under ElasticSearch. Thanks ahead of time for the help!

**Comment by cm-ops** [comment]
Depending on the size of the disk, you might be running into size-based deletion - https://docs.securityonion.net/en/2.4/elasticsearch.html#size-based-index-deletion

**Reply by loganterfehr** [reply]
I thought that too, I had the percentage set to 90% right now. It seems to be clearing at around like 40-50% though. Any other ideas? Note: I did resize the disk from ~150 -> ~400GBs, could the percentage maybe have not updated?
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/8a5049f2-5234-41e5-ad62-0a00c3f0e732)

**Reply by jaypfe** [reply]
By default elastic is set to delete indices once it reaches 50% of the total disk space available.

https://docs.securityonion.net/en/2.4/elasticsearch.html#deleting-indices

"To modify this value, first navigate to [Administration](https://docs.securityonion.net/en/2.4/administration.html#administration) -> Configuration. At the top of the page, click the Options menu and then enable the Show all configurable settings, including advanced settings. option. Then navigate to elasticsearch -> retention -> retention_pct. The change will take effect at the next 15 minute interval. If you would like to make the change immediately, you can click the SYNCHRONIZE GRID button under the Options menu at the top of the page."

**Reply by loganterfehr** [reply]
Thanks Jaypfe! I had it set to 80% for a while now, and I changed it again now to 90%. I'll see if that fixes it. But I had it set to 80% when it was clearing it out at around 50%, so not sure why it is doing that. Any other ideas?

**Reply by cm-ops** [reply]
How big are your indices, `so-elasticsearch-indices-list`? If you are not rolling indices due to size, when the size-based deletion occurs it will remove the datastream and the data in the index.

**Comment by loganterfehr** [comment]
That could definitely be it! How would I go about changing it?
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/2a57a55b-f127-4336-adbc-41a86d5267f7)

**Reply by cm-ops** [reply]
The rollover setting is in `Administration > Configuration > elasticsearch > index_settings > index_name > policy > phases > hot > actions > rollover` where index_name is the name of the index, there is one for time (max_age) and size (max_primary_shard_size).

**Reply by loganterfehr** [reply]
What should I change if I wanted the logs to be stored for 90 days (assuming I have enough storage)

**Reply by cm-ops** [reply]
`Administration -> Configuration -> elasticsearch -> $index -> policy -> phases -> delete -> min_age` That is the time based deletion setting for ILM https://docs.securityonion.net/en/2.4/elasticsearch.html#time-based-index-deletion You would set that to 90 days for the indicies you want to store for 90 days.

**Comment by Phillip-Samson** [comment]
Check your default retention value in Elasticsearch it is probably set to like 50% not sure wy the default is so low
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/30c525f0-5ad5-4187-b05d-28c9b15025c8)
